src/modals/GeolocationModal.jsx

```diff
@@ -45,24 +52,31 @@ function close() {
   ReactDOM.unmountComponentAtNode(document.querySelector('.react_modal__container'));
 }
 
-function open(status, onClose) {
+function open(status, onClose, onClick) {
   ReactDOM.render(
-    <GeolocationModal status={status} onClose={onClose} />,
+    <GeolocationModal status={status} onClose={onClose} onClick={onClick} />,
     document.querySelector('.react_modal__container')
   );
 }
 
-listen('open_geolocate_not_activated_modal', () => open('NOT_ACTIVATED', close));
+listen('open_geolocate_not_activated_modal', () => open('NOT_ACTIVATED', close, close));
 
-listen('open_geolocate_denied_modal', () => open('DENIED', close));
+listen('open_geolocate_denied_modal', () => open('DENIED', close, close));
 
 export async function openAndWaitForClose() {
   return new Promise(resolve => {
-    open('PENDING', () => {
-      close();
-      resolve();
-    });
-  });
+    open(
+      'PENDING',
+      () => {
```

Comment:
you can pass only one onClick handler, and take a status (false or true) as param
```js
  (isAuthorized) => {
    close();
    resolve(isAuthorized); // click "OK": allow native geolocation popup
  }
```